src/streams.rs

```diff
@@ -0,0 +1,174 @@
+                if value.is_err() {
+                    debug!("Non-JSON payload received: {:?}", payload);
+                    continue;
+                }
+                let value = value.unwrap();
+
+                let compiled_schema = compiled_schemas.get_compiled(&topic, &value);
+
+                match compiled_schema {
+                    Ok(schema) => {
+                        let validated = schemas::validate(value, schema);
+
+                        match validated {
+                            Err(validation_error) => {
+                                if let Some(invalid_topic) = &topic.routing.invalid {
+                                    let destination =
+                                        format_routed_topic(&topic.name, invalid_topic);
+
+                                    match serde_json::to_string(&validation_error) {
+                                        Ok(payload) => {
+                                            let message = DispatchMessage {
+                                                destination,
+                                                payload,
+                                            };
+                                            sender.send(message).await;
+                                        }
+                                        Err(e) => {
+                                            error!(
+                                                "Failed to serialize a validation error to JSON: {:?}",
+                                                e
+                                            );
+                                        }
+                                    }
+                                }
+                            }
+
+                            Ok(value) => {
+                                if let Some(valid_topic) = &topic.routing.valid {
+                                    let destination = format_routed_topic(&topic.name, valid_topic);
+                                    let message = DispatchMessage {
+                                        destination,
+                                        payload: value.to_string(),
+                                    };
+
+                                    sender.send(message).await;
+                                } else {
+                                    debug!("No valid topic defined for {}", topic.name);
+                                }
+                            }
+                        }
+                    }
+                    Err(e) => {
+                        error!(
+                            "Error getting compiled schema. Topic: {:?}. Error: {:?}",
+                            &topic, e
+                        );
+                        continue;
+                    }
+                }
+            }
+        }
+    }
+    Ok(())
+}
+
+pub fn create_config(kafka_settings: &HashMap<String, String>) -> ClientConfig {
```

Comment:
Remove unnecessary public access modifier from create_config.
```suggestion
fn create_config(kafka_settings: &HashMap<String, String>) -> ClientConfig {
```